src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.cs

```diff
@@ -21,6 +22,7 @@ public class LibGpiodDriver : UnixDriver
     private readonly ConcurrentDictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
     private readonly ConcurrentDictionary<int, LibGpiodDriverEventHandler> _pinNumberToEventHandler;
     private readonly int _pinCount;
+    private readonly ConcurrentDictionary<int, PinValue> _pinValue;
```

Comment:
I'm looking at this and wondering - perhaps GpioDriver could have virtual `CreateGpioPin` with default implementation - then in here you'd return `LibGpiodPin` and move `CurrentPinValue`, `EventHandler`, `SafeLineHandle` to `LibGpiodPin` - this way you could have just one `ConcurrentDictionary` here and using pin directly would have advantage of not touching ConcurrentDictionary

Reply:
we talked offline and we can do this as a separate change most likely without any API changes